_posts/2016-04-22-responsive-react-components.md

```diff
@@ -0,0 +1,220 @@
+        this.emitChange();
+    }
+
+    getWindowData() {
+        this.width = this.getViewportWidth();
+        this.height = this.getViewportHeight();
+        this.currentViewportSize = this.calculateViewportSize(this.width);
+    }
+
+    getViewportWidth() {
+        return Math.max(document.documentElement.clientWidth, window.innerWidth || 0);
+    }
+
+    getViewportHeight() {
+        return Math.max(document.documentElement.clientHeight, window.innerHeight || 0);
+    }
+
+    calculateViewportSize(width) {
+        if(width < 741) {
+            return ViewportSizes.PALM;
+        } else if (width >= 741 && width < 1025) {
+            return ViewportSizes.LAP;
+        } else if (width >= 1025 && width < 1200) {
+            return ViewportSizes.DESK;
+        } else {
+            return ViewportSizes.DESK_WIDE;
+        }
+    }
+
+    getCurrentViewportSize() {
+        return this.currentViewportSize;
+    }
+}
+```
+
+File: ViewportSizes
+------
+``` javascript
+import keymirror from 'keymirror';
+
+const ViewportSizes = keymirror({
+    PALM: null,
+    LAP: null,
+    DESK: null,
+    DESK_WIDE: null
+});
+
+export default {
+    ViewportSizes
+};
+```
+
+This store holds, calculates and exposes the `currentViewportSize` as a `string` using the `ViewportSizes` constants with which we expose to any components that may be interested using the `getCurrentViewportSize()` function and the `connectToStores` fluxible add-on within a component:
+
+File: AnyComponent
+------
+``` javascript
+import {connectToStores} from 'fluxible-addons-react';
+import WindowStore from 'stores/WindowStore';
+import {ViewportSizes} from 'constants/ViewportSizes';
+
+@connectToStores( [NavigationStore, WindowStore], (context, props) => ({
+    viewportSize: context.getStore(WindowStore).getCurrentViewportSize(),
+}))
+
+class AnyComponent extends React.Component {
+
+    static propTypes = {
+        viewportSize: PropTypes.oneOf(Object.keys(ViewportSizes)).isRequired
+    }
+
+    render() {
+        let size = this.props.viewportSize;
+
+        // This is where the magic happens
+        let menuButton = (size === ViewportSizes.PALM || size === ViewportSizes.LAP) ? <MenuButton/> : null;
+        
+        return (
+            <div>
+                <Search/>
+                {menuButton}
+            </div>
+        )
+    }
+}
+```
+
+As can be seen in the `AnyComponent` component above, it simply listens to changes emitted from the `WindowStore`, compares the current `viewportSize` string with the constants defined in `ViewportSizes` and decides whether to render the `MenuButton` child component or not.
+
+Pairing the Flux architecture with an event manager has allowed the squad to abstract out the actual breakpoint values from each component, this single capability as enabled a great power of rendering anything we like from within a component while making for an easier upgrade path when inevitable "standard" device sizes change. 
```

Comment:
Maybe give this paragraph a title and expand it a little bit since it's the conclusion of the article